### PR TITLE
Clarify PR review and merge criteria

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

My aim is to reduce the number of PRs which have one or more approvals, but are not merged. 

Based on my observations, I believe this is happening because: 
1. It's not clear how many people need to review a PR.
2. Even when a PR is read to be merged, we tend to leave it for the author to merge. 

The changes I'd add to our process are: 

1. PRs should be in draft unless the author is comfortable with it being merged immediately by a reviewer.
2. Code owners who review and approve a PR should either merge it immediately, or indicate what further review they deem necessary (and from whom).
3. **The author of the PR would no longer be the one to merge it.** AFAICT this point is a significant departure from our status quo flow.

### CODE REVIEW

**Comment by maurelian** [comment]
I think 2 more approvals is reasonable for merge.

**Comment by omarish** [comment]
I agree, especially the part about `what further review they deem necessary (and from whom)`.

**Review by omarish — APPROVED** [review]

File: CONTRIBUTING.md
Comment (tynes) [inline CONTRIBUTING.md:26]: Nit: markdown numbered lists should always use `1.` and the rendering process will show incrementing values. This is nice for reducing diffs

File: CONTRIBUTING.md
Comment (tynes) [inline CONTRIBUTING.md:32]: and CI passes

File: CONTRIBUTING.md
Comment (tynes) [inline CONTRIBUTING.md:17]: We should consider adding policy around code owner responding to an opened PR within 2 days, especially when its from an outside contributor.

File: CONTRIBUTING.md
Comment (tynes) [inline CONTRIBUTING.md:14]: We should be explicit around proving context in the PR comment

File: CONTRIBUTING.md
Comment (tynes) [inline CONTRIBUTING.md:46]: We often use merge commits, we do not have the best git hygiene even with it. Standardize around merge commits? Or rebasing into clean commits with comments, this is mostly important for geth/contracts

File: CONTRIBUTING.md
Comment (smartcontracts) [reply CONTRIBUTING.md:17]: Agree. We need a way to deal with draft PRs that just sit as drafts forever.

File: CONTRIBUTING.md
Comment (smartcontracts) [inline CONTRIBUTING.md:44]: There should be a clear point at which the PR can be merged. E.g., if the change requires two approvals and you're about to give the second approval the PR should be merged (by you, immediately).

File: CONTRIBUTING.md
Comment (maurelian) [reply CONTRIBUTING.md:17]: I'm up for that. Is 2 _business days_ reasonable?

Comment (maurelian) [reply CONTRIBUTING.md:17]: > Agree. We need a way to deal with draft PRs that just sit as drafts forever.

@smartcontracts I think that in that case my preference would be the "(simply do not open a PR yet)".

IMO the only good reason for a draft is to get feedback on a WIP. Once the feedback is provided, maybe we should just close it then, along with a warm invitation to reopen when it's ready for final review?

File: CONTRIBUTING.md
Comment (maurelian) [reply CONTRIBUTING.md:44]: I agree, and that's my intent here, with the addition that it should be OK for the second reviewer to indicate that they are not merging it because they want someone else to look at it.

I think this sentence is saying that, but am open to suggestions for improvement.

File: CONTRIBUTING.md
Comment (tynes) [reply CONTRIBUTING.md:19]: We could use a cron job based github action to close stale PRs - https://github.com/marketplace/actions/close-stale-issues
I kind of like having open draft PRs as a way to know what people are working on - if there was a better way to discover that information, I'd be open to it.

>  Is 2 business days reasonable?

Yes

File: CONTRIBUTING.md
Comment (tynes) [reply CONTRIBUTING.md]: What I mean is actually respecting the PR template (filling it out)

**Comment by smartcontracts** [comment]
@maurelian what's left here? Do you need another round of reviews?

File: CONTRIBUTING.md
Comment (maurelian) [reply CONTRIBUTING.md:19]: I created a ticket for that cron job:
https://github.com/ethereum-optimism/roadmap/issues/863

File: CONTRIBUTING.md
Comment (maurelian) [reply CONTRIBUTING.md]: I changed the last point in the list above to:
> Ensure each section of the [Pull Request Template](./PULL_REQUEST_TEMPLATE.md) is filled out.

**Comment by maurelian** [comment]
@smartcontracts I think I've addressed all the good feedback provided by you and @tynes. Since you both took interest, I'd just like to have explicit approval before merging.

File: CONTRIBUTING.md
Comment (tynes) [inline CONTRIBUTING.md:18]: One additional thing that surfaced today was to explicitly delete the boilerplate sections if they do not apply to the PR

File: CONTRIBUTING.md
Comment (tynes) [inline CONTRIBUTING.md:60]: A single...?

Comment (tynes) [reply CONTRIBUTING.md:60]: A single codeowner

File: CONTRIBUTING.md
Comment (maurelian) [reply CONTRIBUTING.md]: Thx, I addressed that here and the [comment](https://github.com/ethereum-optimism/.github/pull/4/files#diff-479755504b0aad0fed26047714591511b4d07c76bc8521171b87a5f16100f850R2) in the PR template

**Pull request merged**
